src/tools/FreeDraw.ts

```diff
@@ -1,47 +1,66 @@
 import { Tool } from '@/tools/Tool'
 import Konva from 'konva'
+import { CanvasElementPayload, CanvasElement } from '@/types/Canvas'
+import uuid from 'uuid'
+import throttle from 'lodash.throttle'
 
 export default class FreeDraw implements Tool {
-  // eslint-disable-next-line no-useless-constructor
+  private line: Konva.Line;
   constructor (public readonly name: string,
                public size: number,
                public colour: string) {
+    this.line = new Konva.Line({ points: [0, 0] })
   }
 
-  action (e: Konva.KonvaPointerEvent, stage: Konva.Stage | null, layer: Konva.Layer): void {
-    // created canvas we can add to layer as "Konva.Image" element
-    const line = new Konva.Line({
-      stroke: this.colour,
-      strokeWidth: Number(this.size),
-      lineJoin: 'round',
-      globalCompositeOperation: 'source-over',
-      points: [e.evt.x, e.evt.y]
-    })
-    layer.add(line)
-
-    const newPoints = line.points().concat([e.evt.x + 3, e.evt.y + 3])
-    line.points(newPoints)
+  mouseDownAction = (e: Konva.KonvaPointerEvent, canvasElementPayload: CanvasElementPayload, layer: Konva.Layer, _socket: WebSocket): void => {
+    canvasElementPayload.data = []
+    canvasElementPayload.data.push({ x: e.evt.x, y: e.evt.y })
+    this.line = this.createElement(canvasElementPayload)
+    layer.add(this.line)
+  }
 
+  mouseMoveAction = throttle((e: Konva.KonvaPointerEvent, canvasElementPayload: CanvasElementPayload, layer: Konva.Layer, _socket: WebSocket): void => {
+    const data = { x: e.evt.x, y: e.evt.y }
+    canvasElementPayload.data.push(data)
+    const newPoints: number[] = this.line.points().concat([data.x, data.y])
+    this.line.points(newPoints)
+    layer.add(this.line)
     layer.batchDraw()
+  }, 10)
 
-    // // Now we need to get access to context element
-    // const context = canvas.getContext('2d')
-    // context.strokeStyle = tool.colour
-    // context.lineJoin = 'round'
-    // context.lineWidth = tool.size
+  mouseUpAction = (e: Konva.KonvaPointerEvent, canvasElementPayload: CanvasElementPayload, layer: Konva.Layer, socket: WebSocket): void => {
+    this.sendToWebsockets(socket, canvasElementPayload)
+  }
 
-    // let pointerPosition = stage.getPointerPosition()
-    // context.globalCompositeOperation = 'source-over'
-    // context.beginPath()
+  renderCanvas = (canvasElement: CanvasElement, layer: Konva.Layer): void => {
+    console.log('layer', layer) // eslint-disable-line
```

Comment:
Are the eslint comments necessary?

Reply:
nope not at all, just didn't like too many warnings when serving the app